Add event type docs (#1457)
yewstack/yew: docs/concepts/html/elements.md

}
```
<!--END_DOCUSAURUS_CODE_TABS-->

## Event Types

| Event name  | `web_sys` Event Type | `stdweb` Event Type |
| ----------- | -------------------- |  ------------------ |
| `onabort` | [Event](https://docs.rs/web-sys/latest/web_sys/struct.Event.html) | [ResourceAbortEvent](https://docs.rs/stdweb/latest/stdweb/web/event/struct.ResourceAbortEvent.html) |
| `onauxclick` | [MouseEvent](https://docs.rs/web-sys/latest/web_sys/struct.MouseEvent.html) | [AuxClickEvent](https://docs.rs/stdweb/latest/stdweb/web/event/struct.AuxClickEvent.html) |
| `onblur` | [FocusEvent](https://docs.rs/web-sys/latest/web_sys/struct.FocusEvent.html) | [BlurEvent](https://docs.rs/stdweb/latest/stdweb/web/event/struct.BlurEvent.html) |
| `oncancel` | [Event](https://docs.rs/web-sys/latest/web_sys/struct.Event.html) | Unsupported |
| `oncanplay` | [Event](https://docs.rs/web-sys/latest/web_sys/struct.Event.html) | Unsupported |
| `oncanplaythrough` | [Event](https://docs.rs/web-sys/latest/web_sys/struct.Event.html) | Unsupported |

> | `onchange` | [ChangeData](https://docs.rs/web-sys/latest/web_sys/struct.ChangeData.html) | [ChangeEvent](https://docs.rs/stdweb/latest/stdweb/web/event/struct.ChangeEvent.html) |

Reviewer: `ChangeData` isn't defined in `web-sys`. Yew uses a special type for this: <https://docs.rs/yew/latest/yew/events/enum.ChangeData.html>.

Author: Fixed it

| `ondragstart` | [DragEvent](https://docs.rs/web-sys/latest/web_sys/struct.DragEvent.html) | [DragStartEvent](https://docs.rs/stdweb/latest/stdweb/web/event/struct.DragStartEvent.html) |
| `ondrop` | [DragEvent](https://docs.rs/web-sys/latest/web_sys/struct.DragEvent.html) | [DragDropEvent](https://docs.rs/stdweb/latest/stdweb/web/event/struct.DragDropEvent.html) |
| `ondurationchange` | [Event](https://docs.rs/web-sys/latest/web_sys/struct.Event.html) | Unsupported |
| `onemptied` | [Event](https://docs.rs/web-sys/latest/web_sys/struct.Event.html) | Unsupported |
| `onended` | [Event](https://docs.rs/web-sys/latest/web_sys/struct.Event.html) | Unsupported |
| `onerror` | [Event](https://docs.rs/web-sys/latest/web_sys/struct.Event.html) | [ResourceErrorEvent](https://docs.rs/stdweb/latest/stdweb/web/event/struct.ResourceErrorEvent.html) |
| `onfocus` | [FocusEvent](https://docs.rs/web-sys/latest/web_sys/struct.FocusEvent.html) | [FocusEvent](https://docs.rs/stdweb/latest/stdweb/web/event/struct.FocusEvent.html) |
| `onformdata` | [Event](https://docs.rs/web-sys/latest/web_sys/struct.Event.html) | Unsupported |

> | `oninput` | [ChangeData](https://docs.rs/web-sys/latest/web_sys/struct.ChangeData.html) | [InputEvent](https://docs.rs/stdweb/latest/stdweb/web/event/struct.InputEvent.html) |

Reviewer: You probably forgot to update the event type here. This should be `InputData` (<https://docs.rs/yew/latest/yew/events/struct.InputData.html>)

Luckily `onchange` and `oninput` are the only event  handlers which wrap the event in a special type.

Author: Fixed that too

}
```
<!--END_DOCUSAURUS_CODE_TABS-->

## Event Types

:::note

> In the following table, "`web-sys` Event Type" should be considered if you're using `yew` with `web-sys` (default); `stdweb` if you use `yew-stdweb`. See [Choose web-sys or stdweb](https://yew.rs/docs/getting-started/choose-web-library) for more information.

Reviewer 2: I don't think the semicolon really helps here (maybe two separate sentences would make it better?). It would be nice if this could be split up a little bit (as part of an ongoing attempt of mine to make diffs readable).

```suggestion
In the following table `web-sys`'s event types should only be used if you're using `yew` with `web-sys`
(this is enabled by default). Use `stdweb`'s event types if you're using the `yew-stdweb` crate. See 
[the documentation page about whether to choose `web-sys` or `stdweb`](https://yew.rs/docs/getting-started/choose-web-library) for more information.
```

Author: I feel like that's a lot unnecessary words and not really concise.
> as part of an ongoing attempt of mine to make diffs readable

What makes the current sentence unreadable?

Reviewer 2: > What makes the current sentence unreadable?

It's not the actual sentence – it's just handy to cap the line lengths so that when it comes to reviewing later changes it's easy to see exactly what's been changed.

I'm not particularly concerned about the exact wording, but the current sentence has a few problems:
* "Event Type" isn't a proper noun so should be lower case. I think it's better to say something like "event types from `web-sys`" or `web-sys`'s event types rather than `web-sys` event type (there are multiple event types so I think that should be plural).
* The semicolon doesn't make gramatical sense – they can't be used to connect clauses which can't stand independent of each other (i.e. they must be complete sentences).
* "See Choose `web-sys` or `std-web`" doesn't really fit very well. "Choose" definitely has to be lower case there and I think it should be in the present participle rather than in the infinitive (i.e. "choosing" rather than "choose").

Author: Understandable, I have merged your suggestion.

:::

:::note

> All the event types mentioned in the following table are re-exported under `yew::events`.

Reviewer 2: ```suggestion
All the event types mentioned in the following table are re-exported under `yew::events`. Using the types from
`yew::events` makes it easier to ensure version compatibility than if you were to manually include `web-sys` 
or `stdweb` as dependencies in your crate because you won't end up using a version which conflicts with
the version Yew specifies. 
```

| `oncancel` | [Event](https://docs.rs/web-sys/latest/web_sys/struct.Event.html) | Unsupported |
| `oncanplay` | [Event](https://docs.rs/web-sys/latest/web_sys/struct.Event.html) | Unsupported |
| `oncanplaythrough` | [Event](https://docs.rs/web-sys/latest/web_sys/struct.Event.html) | Unsupported |

> | `onchange` | [ChangeData](https://docs.rs/yew/latest/yew/events/enum.ChangeData.html) | [ChangeEvent](https://docs.rs/stdweb/latest/stdweb/web/event/struct.ChangeEvent.html) |

Reviewer: ```suggestion
| `onchange` | [ChangeData](https://docs.rs/yew/latest/yew/events/enum.ChangeData.html) | [ChangeData](https://docs.rs/yew-stdweb/latest/yew_stdweb/events/enum.ChangeData.html) |
```

| `onerror` | [Event](https://docs.rs/web-sys/latest/web_sys/struct.Event.html) | [ResourceErrorEvent](https://docs.rs/stdweb/latest/stdweb/web/event/struct.ResourceErrorEvent.html) |
| `onfocus` | [FocusEvent](https://docs.rs/web-sys/latest/web_sys/struct.FocusEvent.html) | [FocusEvent](https://docs.rs/stdweb/latest/stdweb/web/event/struct.FocusEvent.html) |
| `onformdata` | [Event](https://docs.rs/web-sys/latest/web_sys/struct.Event.html) | Unsupported |

> | `oninput` | [ChangeData](https://docs.rs/yew/latest/yew/events/struct.InputData.html) | [InputEvent](https://docs.rs/stdweb/latest/stdweb/web/event/struct.InputEvent.html) |

Reviewer: ```suggestion
| `oninput` | [InputData](https://docs.rs/yew/latest/yew/events/struct.InputData.html) | [InputData](https://docs.rs/yew-stdweb/latest/yew_stdweb/events/struct.InputData.html) |
```
